tests/integration_tests/tests/agentless_tests/test_blueprint_upload.py

```diff
@@ -0,0 +1,242 @@
+########
+# Copyright (c) 2020 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import copy
+
+import requests
+import requests.status_codes
+
+from cloudify.models_states import BlueprintUploadState
+
+from integration_tests import AgentlessTestCase
+from integration_tests.tests.utils import get_resource as resource
+from integration_tests.tests.utils import wait_for_blueprint_upload_to_end
+
+
+class BlueprintUploadTest(AgentlessTestCase):
+    def test_blueprint_upload(self):
+        blueprint_id = 'bp'
+        blueprint_filename = 'empty_blueprint.yaml'
+        self.client.blueprints.upload(
+            resource('dsl/{}'.format(blueprint_filename)),
+            entity_id=blueprint_id
+        )
+        wait_for_blueprint_upload_to_end(blueprint_id, self.client)
+        blueprint = self.client.blueprints.get(blueprint_id)
+
+        self.assertEqual(blueprint['state'], BlueprintUploadState.UPLOADED)
+        self.assertEqual(blueprint.main_file_name, blueprint_filename)
+        self.assertNotEqual(blueprint.plan, None)
+        self._verify_blueprint_files(blueprint_id, blueprint_filename)
+
+    def test_blueprint_upload_from_url(self):
```

Comment:
This test has the same flow as other tests in this class. I think it would better if you took out the logic to an auxiliary function.

Reply:
done